python/amico/core.py

```diff
@@ -294,6 +299,9 @@ def fit( self ) :
                         if b0 > 1e-3 :
                             y = y / b0
 
+                    if self.get_config('doMergeB0') and self.scheme.b0_count > 0:
+                        y = np.hstack((np.mean( y[self.scheme.b0_idx] ),y[self.scheme.dwi_idx]))
```

Comment:
The advantage of cropping the signal here is that the "doMergeB0" flag can be turned on and off and the data reprocessed quickly. Cropping the whole DWI volume and the KERNELS beforehand would require to go through the whole load_data() and load_kernels() procedures when the flag is switched, which can be time consuming, specially regarding the NODDI model.